### PR TITLE
feat(draw-sw): add simple Helium acceleration

### PR DESCRIPTION
**Pull request by PeterBee97**
### Description of the feature or fix
Similar to Neon, this patch added Helium acceleration to all blending functions with minimal code. Still, the assembly code will produce slightly different result (normally 1bit error) from C rendering results due to difference in implementation, but visually similar to the eye:
cbimage (8)
The argb8888 dst is also supported with parallel division by integer newton iteration method. Thanks to tail predication and scatter-gathering functions, the code is conciser than Neon version and should also be more efficient. However, speed is not yet tested as the development was done on QEMU.
### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by PeterBee97** [comment]
Should be able to work together with arm2d. @GorgonMeducer please help review and test it! :)

**Comment by PeterBee97** [comment]
> Do you consider adding other Helium acceleration? For example, for transform, rotate screen etc.
> 
I'll look into transforms later.

> By the way, would you please provide a macro option to only use arm-2d acceleration without the ASM version you added? (There is an option only to use your helium assembly version without arm-2d).
> 
I've moved arm2d header inclusion out of lv_blend_helium.h since arm2d already has its individual config LV_USE_DRAW_ARM2D_SYNC, so now the two accelerator can be toggled independently.

**Comment by kisvegabor** [comment]
Amazing! However it's not entirely clear how to select between pure ASM and Arm2D. I guess you need to set `LV_USE_DRAW_SW_ASM LV_DRAW_SW_ASM_HELIUM` in both cases. And if you enable `LV_USE_DRAW_ARM2D_SYNC` Arm2D will be used, if it's disabled the plain ASM?

**Comment by PeterBee97** [comment]
> Amazing! However it's not entirely clear how to select between pure ASM and Arm2D. I guess you need to set `LV_USE_DRAW_SW_ASM LV_DRAW_SW_ASM_HELIUM` in both cases. And if you enable `LV_USE_DRAW_ARM2D_SYNC` Arm2D will be used, if it's disabled the plain ASM?

That's what I previously thought, but since arm2d cannot override all of asm functions right now enabling LV_DRAW_SW_ASM_HELIUM means asm functions cannot be shutdown entirely (which is bad for Gabriel's test). Having the two options decoupled means we can have arm2d, or asm, or both (where asm is the backup). Or is there better idea to let arm2d work by itself? Need some help here. @kisvegabor

**Comment by GorgonMeducer** [comment]
@PeterBee97  @kisvegabor 
> That's what I previously thought, but since arm2d cannot override all of asm functions right now enabling LV_DRAW_SW_ASM_HELIUM means asm functions cannot be shutdown entirely (which is bad for Gabriel's test). 

The original solution is the best one. I only need a macro option for testing purpose only.

**Comment by PeterBee97** [comment]
Added macro ```USE_NATIVE_ASSEMBLY``` to disable functions internally.

**Review by kisvegabor — APPROVED** [review]
Clear, looks good to me. We need to document these somewhere though. 

I'll also do some tests today.

File: src/draw/sw/blend/helium/lv_blend_helium.h [hunk 1]
```diff
@@ -0,0 +1,1310 @@
...
+#include "../../../../lv_conf_internal.h"
+
+/* detect whether helium is available based on arm compilers' standard */
+#if defined(__ARM_FEATURE_MVE) && __ARM_FEATURE_MVE
+
+#ifdef LV_DRAW_SW_HELIUM_CUSTOM_INCLUDE
+#include LV_DRAW_SW_HELIUM_CUSTOM_INCLUDE
+#endif
+
+#define USE_NATIVE_ASSEMBLY 1
```
Comment (kisvegabor) [inline src/draw/sw/blend/helium/lv_blend_helium.h]: It should come from a compiler define.

Please also change the name to `LV_USE_NATIVE_ASSEMBLY`

**Review by kisvegabor — CHANGES_REQUESTED** [review]
I've just found something.

File: src/draw/sw/blend/helium/lv_blend_helium.h (hunk 1 above)
Comment (PeterBee97) [reply src/draw/sw/blend/helium/lv_blend_helium.h]: Sure, moved to lv_conf.

**Comment by GorgonMeducer** [comment]
Looks good for me.

**Review by kisvegabor — APPROVED** [review]
At this point I'm not sure ` #define  LV_USE_DRAW_SW_ASM     LV_DRAW_SW_ASM_...` is still required. As ASM implementation can overlap probably we should enable them differently. 

Let's improve it in an other PR later.

**Pull request merged**